langchain4j-elasticsearch/src/test/java/dev/langchain4j/store/embedding/elasticsearch/ElasticsearchEmbeddingStoreRemoveIT.java

```diff
@@ -0,0 +1,198 @@
+        List<String> matchingIds = matches.stream().map(EmbeddingMatch::embeddingId).collect(Collectors.toList());
+        assertThat(matchingIds).hasSize(1);
+        assertThat(matchingIds).containsExactly(id);
+    }
+
+    @Test
+    void remove_all_by_ids_null() {
+        assertThatThrownBy(() -> embeddingStore.removeAll((Collection<String>) null))
+                .isExactlyInstanceOf(IllegalArgumentException.class)
+                .hasMessage("ids cannot be null or empty");
+    }
+
+    @Test
+    void remove_all_by_filter() {
+        // given
+        Metadata metadata = Metadata.metadata("id", "1");
+        TextSegment segment = TextSegment.from("matching", metadata);
+        Embedding embedding = embeddingModel.embed(segment).content();
+        embeddingStore.add(embedding, segment);
+        awaitUntilPersisted();
+        Embedding embedding2 = embeddingModel.embed("hello2").content();
+        Embedding embedding3 = embeddingModel.embed("hello3").content();
+
+        String id2 = embeddingStore.add(embedding2);
+        String id3 = embeddingStore.add(embedding3);
+        awaitUntilPersisted();
+
+        // when
+        embeddingStore.removeAll(metadataKey("id").isEqualTo("1"));
+        awaitUntilPersisted();
+
+        // then
+        EmbeddingSearchRequest request = EmbeddingSearchRequest.builder()
+                .queryEmbedding(embedding)
+                .maxResults(10)
+                .build();
+        List<EmbeddingMatch<TextSegment>> matches = embeddingStore.search(request).matches();
+        List<String> matchingIds = matches.stream().map(EmbeddingMatch::embeddingId).collect(Collectors.toList());
+        assertThat(matchingIds).hasSize(2);
+        assertThat(matchingIds).containsExactly(id2, id3);
+    }
+
+    @Test
+    void remove_all_by_filter_not_matching() {
+        // given
+        Embedding embedding = embeddingModel.embed("hello").content();
+        Embedding embedding2 = embeddingModel.embed("hello2").content();
+        Embedding embedding3 = embeddingModel.embed("hello3").content();
+
+        embeddingStore.add(embedding);
+        embeddingStore.add(embedding2);
+        embeddingStore.add(embedding3);
+        awaitUntilPersisted();
+
+        // when
+        embeddingStore.removeAll(metadataKey("unknown").isEqualTo("1"));
+        awaitUntilPersisted();
+
+        // then
+        EmbeddingSearchRequest request = EmbeddingSearchRequest.builder()
+                .queryEmbedding(embedding)
+                .maxResults(10)
+                .build();
+        List<EmbeddingMatch<TextSegment>> matches = embeddingStore.search(request).matches();
+        List<String> matchingIds = matches.stream().map(EmbeddingMatch::embeddingId).collect(Collectors.toList());
+        assertThat(matchingIds).hasSize(3);
+    }
+
+    @SneakyThrows
+    protected void awaitUntilPersisted() {
+        // TODO: not sure if it is ok, 1 sec was not enough
```

Comment:
please take a look at this, it was not working with just 1 second like in ElasticsearchEmbeddingStoreIT